src/chum.cpp

```diff
@@ -450,3 +454,86 @@ void Chum::setStatus(QString status) {
     m_status = status;
     emit statusChanged();
 }
+
+void Chum::getChumCache()
+{
+    if (!m_busy) {
+        m_busy = true;
+        emit busyChanged();
+    }
+
+    //% "Receiving the chum package cache"
+    setStatus(qtTrId("chum-update-cache"));
+
+    QNetworkAccessManager *netManager = new QNetworkAccessManager();
+
+    QObject::connect(netManager,&QNetworkAccessManager::finished,[=](QNetworkReply *reply) {
+        if (reply->error()){
+            qDebug() << "Unable to get package build times" << reply->errorString();
+        }
+        else {
+            m_chumPackageCache = QJsonDocument::fromJson(reply->readAll());
+            qDebug() << "Received chum package cache";
+        }
+        m_busy = false;
+        //% "Loading SailfishOS:Chum repository"
+        setStatus(qtTrId("chum-load-repositories"));
+        m_ssu.loadRepos();
+    });
+
+    // Start the network request
+    QNetworkRequest request=QNetworkRequest(QUrl(apiUrl));
+    m_busy = true;
+    emit busyChanged();
+    netManager->get(request);
+}
+
+QDateTime Chum::findPackageMTime(const QString &rpm) const
+{
+    qDebug() << Q_FUNC_INFO << rpm << Chum::instance()->repoName() << Chum::instance()->repoVersion();
+
```

Comment:
I wonder whether there is a better way of doing it.  In principle, we have for each package its name (ex 'harbour-advanced-camera' or 'libscrypt-devel'). At the same time, we can expose URL of active Chum repository. 

For exposing Chum repo URL, we would have We would just need to add one more property (repoUrl?) to SSU and fill it up based on m_repos (see https://github.com/sailfishos-chum/sailfishos-chum-gui/blob/main/src/ssu.cpp#L60 and below). Thus, we could get current repo using m_ssu.repoUrl() and get https://repo.sailfishos.org/obs/sailfishos:/chum:/testing/4.5.0.16_aarch64/

After that, it is possible to condition URL into `testing/4.5.0.16_aarch64` or `chum/4.5.0.16_aarch64` and use those in JSON. 

When generating JSON, we can actually forget about "projects" and immediately link those to "packages". So, that would be a task of backend solution. Ideally, we should have JSON as follows:

```json
{
"testing/4.5.0.16_aarch64": {
"harbour-advanced-camera": { "mtime": "1634426517" },
"libscrypt-devel":  { "mtime": "1634426517" }
}
}
```

That way it can be extended by more details if needed and would ensure that we have data accessible easily from GUI.